Core/DisconnectMeStore.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  DisconnectMeStore.swift
+//  DuckDuckGo
+//
+//  Created by Christopher Brind on 12/09/2017.
+//  Copyright © 2017 DuckDuckGo. All rights reserved.
+//
+
+import Foundation
+
+class DisconnectMeStore {
+
+    static let shared = DisconnectMeStore()
+
+    var hasData: Bool {
+        return jsonString != "[]"
+    }
+
+    private(set) var jsonString: String = "[]"
+
+    private var persistenceLocation: URL {
+        get {
+            let documentPath = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0] as String
+            let path = URL(fileURLWithPath: documentPath, isDirectory: true)
+            return path.appendingPathComponent("disconnectme.json")
+        }
+    }
+
+    private init() {
+        _ = try? parse(data: Data(contentsOf: persistenceLocation))
+    }
+
+    func persist(data: Data) throws -> Int {
+        let count = try parse(data: data)
+        try! data.write(to: persistenceLocation, options: .atomic)
```

Comment:
Do you want to crash if this fails?

Reply:
I can have it throw the exception instead. Will update in next PR.